terralab/utils.py

```diff
@@ -36,3 +41,50 @@ def wrapper(*args, **kwargs):
             exit(1)
 
     return wrapper
+
+
+def process_json_to_dict(json_data) -> typing.Optional[dict]:
+    """Process the given JSON string and return a dictionary.
+    Returns None if the input string is not able to be parsed to a dictionary."""
+    try:
+        data = json.loads(json_data)
+        if not (isinstance(data, dict)):
+            raise TypeError
+        LOGGER.debug(f"Processed inputs: {data}")
+        return data
+    except (json.JSONDecodeError, TypeError):
+        LOGGER.error("Input string not parsable to a dictionary.")
+        return None
+
+
+def is_valid_local_file(local_file_path: str) -> bool:
+    """Validate that the provided local file path exists."""
+    return True if os.path.exists(local_file_path) else False
+
+
+def validate_pipeline_inputs(pipeline_name: str, inputs_dict: dict):
+    """Check that all required user inputs are present, and that all file inputs are valid."""
+    # retrieve required pipeline inputs; note currently this endpoint only returns required inputs
+    pipeline_info: PipelineWithDetails = get_pipeline_info(pipeline_name)
+
+    input_error_messages = []
+    for (
+        input_info
+    ) in pipeline_info.inputs:  # input_info is PipelineUserProvidedInputDefinition
+        input_name: str = input_info.name
```

Comment:
any particular reason we do typing for some variables and not others?

Reply:
just cause it's not required in Python and I only sometimes remember to do it